File: Algorithm/LeetCode/daily/2024_09/26_2535_mid.ts
/**
 * @Time 2024/09/26 14:23
 * @Author  : https://github.com/hakusai22
 */

function differenceOfSum(nums: number[]): number {
    let ans = 0
    for (let num of nums) {
        ans += num
        while (num) {
            ans -= num % 10
            num = Math.floor(num / 10)
        }
    }
    return ans
}

File: Algorithm/LeetCode/daily/2024_09/26_2535_mid.go
package _024_09

/*
    @Author  : https://github.com/hakusai22
    @Time    : 2024/09/26 14:23
    @题目     : https://leetcode.cn/problems/difference-between-element-sum-and-digit-sum-of-an-array/description/
    @参考     : https://leetcode.cn/problems/difference-between-element-sum-and-digit-sum-of-an-array/solutions/2062785/bian-li-by-endlesscheng-hi82/
    @时间复杂度: O(nlogU)，其中 n 为 nums 的长度，U=max(nums)。
    @空间复杂度:

 数据范围:

*/

func differenceOfSum(nums []int) (ans int) {
	for _, x := range nums {
		ans += x
		for x > 0 {
			ans -= x % 10
			x /= 10
		}
	}
	return ans
}
